Update docker-compose.yml to reference v1 tag, instead of defaulting to latest.

josh-padnick opened the pull request:
Simple change, but makes the `docker-compose.yml` work out of the box.

A few helpful tips with this:

1. As of Oct 12, 2016, `docker-compose up` is really slow on Docker beta for Mac. The fix is [here](https://github.com/docker/compose/issues/3419#issuecomment-221793401).

2. There's a known issue where the first run of `docker-compose up` is successful, but the second run fails because the files in `./rails-frontend/tmp` don't automatically get deleted when you `CTRL+C` docker-compose.

@brikis98 Any thought on a way to handle the second issue?

Also, I've pushed `v2` of the `sinatra-backend` to https://hub.docker.com/r/gruntwork/sinatra-backend.

brikis98 commented:
@josh-padnick I think it would be better to leave `docker-compose.yml` on the `latest` tag (i.e. no tag). That way, whenever you build the image locally, it picks it up automatically, and you don't have to keep updating that file to a new version.

brikis98 commented:
> There's a known issue where the first run of docker-compose up is successful, but the second run fails because the files in ./rails-frontend/tmp don't automatically get deleted when you CTRL+C docker-compose.

Fails with what error?

josh-padnick commented:
> I think it would be better to leave docker-compose.yml on the latest tag (i.e. no tag). 

I do see your point, although since it's a demo repo I figured `v1` was more clear since now users have to explicitly change from `latest` and there's some ambiguity about which one that is: (does that represent `v1` or `v2`?).

> Fails with what error?

```
docker-compose up
<works as expected>

CTRL-C
docker-compose up
...
rails_frontend_1   | A server is already running. Check /usr/src/app/tmp/pids/server.pid.
...
```

The issue is that `./rails-frontend/tmp` doesn't get deleted on `CTRL-C`, and this line in `docker-compose.yml`:

``` yml
volumes:
    - ./rails-frontend:/usr/src/app
```

mounts the volume into the container. Ideally, sending a `SIGTERM` to `docker-compose` would trigger some graceful shutdown procedure by rails. Come to think of it, does `CTRL-C` trigger a graceful shutdown in docker-compose? It seems like sometimes yes, sometimes no, to be honest.

brikis98 commented:
> I do see your point, although since it's a demo repo I figured v1 was more clear since now users have to explicitly change from latest and there's some ambiguity about which one that is: (does that represent v1 or v2?).

Not sure I follow. The dev environment should always be on "latest". Prod environments should always be on specific versions. Every CI build should tag an image twice: once with a new version (e.g. v2, v3, v4, etc) and once with latest.

brikis98 commented:
> rails_frontend_1   | A server is already running. Check /usr/src/app/tmp/pids/server.pid.

That's weird. I wonder why I never saw that error? I ran and restarted these containers a bunch of times while putting this talk together... You're not somehow overriding `RAILS_ENV` to `production` are you?

brikis98 commented:
Oh, actually I know why I wasn't seeing this. I'm using [docker-osx-dev](https://github.com/brikis98/docker-osx-dev) so my mounted volumes are "one-way" only. That is, changes on my host OS get copied into the Docker container, but not vice versa. I bet you're using Docker's native volume mounting (which, last I checked, was still painfully slow). As a result, the PID file gets copied back to your host OS, and it's there when you restart. 

I wonder if adding that file to a `.dockerignore` would work? If not, see this SO thread: http://stackoverflow.com/a/38732187/483528

josh-padnick commented:
@brikis98 Ok, we should be good to go here. Made the following updates:
- Added a `docker-entrypoint.sh` script that deletes the `/tmp/.../server.pid` file before booting up.
- Reverted `docker-compose.yml` to use latest.
- Pushed updated `v1`/`latest` and `v2` images to Docker Hub.

brikis98 commented:
LGTM, thx!
